### PR TITLE
RAVTDBQ_366_Regexp_Replace_With_Null_Argument

### PR DESCRIPTION
**Pull request by Pritam-Gote**
Regexp_Replace_With_Null_Argument

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -13895,4 +13895,37 @@ private RexNode createRexForQuantile(RelBuilder builder) {
     assertThat(toSql(root, DatabaseProduct.SNOWFLAKE.getDialect()), isLinux(expectedSnowflakeSql));
   }
 
+  @Test public void testForRegexpReplaceWithReplaceStringAsNull() {
+    final RelBuilder builder = relBuilder();
+    final RexNode regexpReplaceRex = builder.call(SqlLibraryOperators.REGEXP_REPLACE,
+        builder.literal("Calcite"), builder.literal("ac"), builder.literal(null),
```
Comment (Mahesh-Raut) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:13901]: Please try to add a test case scenario similar to our Canonical at the mig side for the BigQuery SQL Dialect.

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 2]
```diff
@@ -1751,6 +1752,25 @@ private void unparseRegexFunctionsOperands(SqlWriter writer, int leftPrec, int r
...
     }
   }
+  /**
+   * This method is to unparse the REGEXP_REPLACE operands.
+   */
+  private void unparseRegexpReplaceFunctionOperands(SqlWriter writer, int leftPrec, int rightPrec,
+      int indexOfRegexOperand, List<SqlNode> operandList) {
+    int operandListSize = operandList.size();
+    for (int index = 0; index < operandListSize; index++) {
+      if (index < 3) {
```
Comment (Mahesh-Raut) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1762]: Please attempt to optimize this code and avoid the use of any type of loop.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (Pritam-Gote) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:13912]: Done

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 2 above)
Comment (Pritam-Gote) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: Done

**Comment by Mahesh-Raut** [comment]
Changes look good to me. 
L1 is done.

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Pull request merged**